src/com/group6/views/RegisterView.java

```diff
@@ -0,0 +1,132 @@
+     * It will ask the user to input their role, personal details, and then create the appropriate user.
+     * After successful registration, it returns to the main menu.
+     *
+     * @return A MenuView instance, which represents the main menu after registration.
+     */
+    public View run() {
+        System.out.println("\n--- User Registration ---");
+
+        // 1. Choose role
+        UserRole role = promptRole();
+
+        // 2. Common inputs
+        String name = prompt("Enter full name: ");
+
+        String nric;
+        while (true) {
+            nric = prompt("Enter NRIC (e.g. S1234567A): ").toUpperCase();
+            if (!validateUtils.isValidNRIC(nric)) {
+                System.out.println("❌ Invalid NRIC format.");
+            } else if (userManager.getUserByNRIC(nric).isPresent()) {
+                System.out.println("❌ NRIC already registered.");
+            } else {
+                break;
+            }
+        }
+
+        int age;
+        while (true) {
+            try {
+                age = Integer.parseInt(prompt("Enter age: "));
+                if (age < 18) {
+                    System.out.println("❌ Age must be 18 or above.");
+                } else {
+                    break;
+                }
+            } catch (NumberFormatException e) {
+                System.out.println("❌ Please enter a valid number.");
+            }
+        }
+
+        UserMaritalStatus maritalStatus;
+        while (true) {
+            System.out.print("Enter marital status (Single/Married): ");
+            String statusInput = scanner.nextLine().trim().toUpperCase();
+            if (statusInput.equals("SINGLE")) {
+                maritalStatus = UserMaritalStatus.SINGLE;
+                break;
+            } else if (statusInput.equals("MARRIED")) {
+                maritalStatus = UserMaritalStatus.MARRIED;
+                break;
+            } else {
+                System.out.println("❌ Invalid input. Please enter 'Single' or 'Married'.");
+            }
+        }
+
+        String password = prompt("Create password: ");
+        String userId = UUID.randomUUID().toString();
+
+        // 3. Create appropriate user
+        User newUser = switch (role) {
+            case APPLICANT -> new Applicant(userId, nric, age, maritalStatus, password);
+            case OFFICER -> new HDBOfficer(userId, nric, age, maritalStatus, password);
+            case MANAGER -> new HDBManager(userId, nric, age, maritalStatus, password);
+        };
+
+        // 4. Save
+        userManager.registerUser(newUser);
+
+        System.out.println("✅ Registration successful as " + role + "! You can now log in.");
+        return new MenuView(scanner, userManager);
+    }
+
+    /**
+     * Prompts the user to select a role for registration.
+     *
+     * @return The UserRole selected by the user (either Applicant, Officer, or Manager).
+     */
+    private UserRole promptRole() {

```

Comment:
Hmmm should we do this or should we assume that all registrations are for applicants?